config/src/main/java/com/typesafe/config/impl/ConfigNodeObject.java

```diff
@@ -0,0 +1,127 @@
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigSyntax;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+final class ConfigNodeObject extends ConfigNodeComplexValue {
+    ConfigNodeObject(Collection<AbstractConfigNode> children) {
+        super(children);
+    }
+
+    protected ConfigNodeObject changeValueOnPath(Path desiredPath, AbstractConfigNodeValue value) {
+        ArrayList<AbstractConfigNode> childrenCopy = new ArrayList(super.children);
+        // Copy the value so we can change it to null but not modify the original parameter
+        AbstractConfigNodeValue valueCopy = value;
+        for (int i = super.children.size() - 1; i >= 0; i--) {
+            if (!(super.children.get(i) instanceof ConfigNodeField)) {
+                continue;
+            }
+            ConfigNodeField node = (ConfigNodeField)super.children.get(i);
+            Path key = node.path().value();
+            if (key.equals(desiredPath)) {
+                if (valueCopy == null)
+                    childrenCopy.remove(i);
+                else {
+                    childrenCopy.set(i, node.replaceValue(value));
+                    valueCopy = null;
+                }
+            } else if (desiredPath.startsWith(key)) {
+                if (node.value() instanceof ConfigNodeObject) {
+                    Path remainingPath = desiredPath.subPath(key.length());
+                    childrenCopy.set(i, node.replaceValue(((ConfigNodeObject)node.value()).changeValueOnPath(remainingPath, valueCopy)));
+                    if (valueCopy != null && node.render() != super.children.get(i).render())
```

Comment:
I guess the `!=` here should be `equals`, but I would think if the renders are equal the original nodes are also equal ... if you want to implement equals() with render(), you could do that in AbstractConfigNode itself perhaps